core/trino-main/src/main/java/io/trino/server/security/ServerSecurityModule.java

```diff
@@ -77,6 +77,10 @@ protected void setup(Binder binder)
             configBinder(binder).bindConfig(PasswordAuthenticatorConfig.class);
             binder.bind(PasswordAuthenticatorManager.class).in(Scopes.SINGLETON);
         }));
+        install(authenticatorModule("header-based", HeaderBasedAuthenticator.class, headerBinder -> {
+            configBinder(binder).bindConfig(HeaderBasedAuthenticatorConfig.class);
```

Comment:
These should use `headerBinder`. The one for `PasswordAuthenticator` above is actually wrong.